### PR TITLE
Feat/signin

### PR DESCRIPTION
**Pull request by ClaireE93**
Create signin popup that stores user and allows for user to add coins to their list which updates "my news".

### CODE REVIEW

File: src/components/News.jsx [hunk 1]
```diff
@@ -35,21 +45,25 @@ class News extends React.Component {
       curData: [],
```
Comment (nsaboo) [inline src/components/News.jsx:45]: default this.state not defined in constructor

File: src/components/News.jsx [hunk 2]
```diff
@@ -35,21 +45,25 @@ class News extends React.Component {
...
+    this.setTrendingData(this.props.coin);
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.coin !== nextProps.coin) {
-      this.setData(nextProps.coin);
+    if (this.props.coin !== nextProps.coin && this.state.trendingSelected) {
+      this.setTrendingData(nextProps.coin);
+    }
+    if (!checkListEquality(this.props.list, nextProps.list)) {
```
Comment (nsaboo) [inline src/components/News.jsx:61]: if props are made state variables inside this component, then setState will handle the deep Comparison for us? So can we make props as state variables here?

Comment (nsaboo) [reply src/components/News.jsx:61]: I guess then we won't require componentWillRecieveProps too along with setTrendingData etc..?

**Review by nsaboo — COMMENTED** [review]
please correct me with the review comments, I might be wrong, though let's discuss..

File: src/components/News.jsx (hunk 2 above)
Comment (ClaireE93) [reply src/components/News.jsx:61]: My understanding is that setting props equal to state is seen as a bad practice (anti-pattern) for react:
https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e

You're supposed to use lifecycle functions like componentWillReceiveProps to monitor prop changes.

**Review by nsaboo — APPROVED** [review]

**Comment by nsaboo** [comment]
awesome, looks good to me

**Pull request merged**